### PR TITLE
Add purchase button and confirmation

### PR DESCRIPTION
**Pull request by oy-gen**
- added purchase button and confirmation.
  - made Nav and Buttons more modular
  - cleaned-up useStore

### CODE REVIEW

File: components/Buttons.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    color: darkgrey;
+    background-color: grey;
+  }
+`;
+
+export const ButtonL = styled.button`
+  grid-column: 2/4;
+  background-color: ${props => props.background};
+  color: ${props => props.textcolor};
+`;
```
Comment (F-Kirchhoff) [inline components/Buttons.js]: Couldn't you comine these two buttons into one by introducing another prop called size?

File: components/Modal.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+const Modal = ({ show, onClose, children }) => {
+  const [isBrowser, setIsBrowser] = useState(false);
+
+  useEffect(() => {
+    setIsBrowser(true);
+  }, []);
+
+  const handleCloseClick = event => {
+    event.preventDefault();
```
Comment (F-Kirchhoff) [inline components/Modal.js]: do you need to prevent some default here? It is not inside a form.

File: components/Modal.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import ReactDOM from 'react-dom';
+import styled from 'styled-components';
+import Link from 'next/link';
+
+const Modal = ({ show, onClose, children }) => {
+  const [isBrowser, setIsBrowser] = useState(false);
+
+  useEffect(() => {
+    setIsBrowser(true);
+  }, []);
```
Comment (F-Kirchhoff) [inline components/Modal.js:11]: This is a bit hacky for disabling the serverside rendering of your modal. You should use a dynamic import when using the modal instead. Here is a link where you can find out how it works:
https://nextjs.org/docs/advanced-features/dynamic-import

File: components/Modal.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+const StyledModal = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  position: relative;
+  margin: 3rem 1rem;
+  background: white;
+  width: 90%;
```
Comment (F-Kirchhoff) [inline components/Modal.js]: maybe you can give your overlay a padding instead of this percentage?

File: components/Modal.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  width: 1.6rem;
+  color: var(--text-lightcolor);
+`;
+
+const StyledModalOverlay = styled.div`
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
```
Comment (F-Kirchhoff) [inline components/Modal.js:46]: What do you need this for?

File: components/Buttons.js (hunk 1 above)
Comment (F-Kirchhoff) [reply components/Buttons.js]: With your grid-column assignment the buttons are not really reuseable. I think you don't need it.

File: components/NavWrapper.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  }
+`;
+
+const NavElementWrapper = styled.div`
+  display: grid;
+  align-items: center;
+  grid-template-columns: 1fr 2fr;
+  width: 800px;
+  @media (max-width: 600px) {
+    grid-template-columns: 136px 1fr;
```
Comment (F-Kirchhoff) [inline components/NavWrapper.js]: this exact value for the column width might lead to a weird layout, I am sure you can find a more elegant solution for that.

File: components/ProductCard.js [hunk 7]
```diff
@@ -60,7 +60,7 @@ export default function ProductCard({ product }) {
 
 const StyledCard = styled.div`
   display: grid;
-  grid-template-columns: 200px 1fr;
+  grid-template-columns: 1fr 2fr;
```
Comment (F-Kirchhoff) [inline components/ProductCard.js]: this is better!

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Eugen, pretty nice. I have some suggestions for your styling and about the client side rendering of your modals, please have a look into it.

File: components/Buttons.js (hunk 1 above)
Comment (oy-gen) [reply components/Buttons.js]: > Couldn't you comine these two buttons into one by introducing another prop called size?
DONE!

> With your grid-column assignment the buttons are not really reuseable. I think you don't need it.
I would like to keep it, the grid layout will be used on other pages as well.

File: components/Modal.js (hunk 5 above)
Comment (oy-gen) [reply components/Modal.js:46]: the dark background needs that to cover the whole screen

File: components/Modal.js (hunk 3 above)
Comment (oy-gen) [reply components/Modal.js:11]: I think i need help with that

**Pull request merged**